OpenRA.Game/Sound/Sound.cs

```diff
@@ -348,9 +350,27 @@ public float VideoSeekPosition
 			var name = prefix + clip + suffix;
 
 			if (!string.IsNullOrEmpty(name) && (p == null || p == p.World.LocalPlayer))
-				soundEngine.Play2D(sounds[name],
-					false, relative, pos,
-					InternalSoundVolume * volumeModifier, attenuateVolume);
+			{
+				if (id != 0)
+				{
+					if (currentSounds.ContainsKey(id))
+					{
+						soundEngine.StopSound((ISound)currentSounds[id]);
+						currentSounds.Remove(id);
+					}
+
+					ISound sound = soundEngine.Play2D(sounds[name],
+						false, relative, pos,
+						InternalSoundVolume * volumeModifier, attenuateVolume);
+						currentSounds.Add(id, sound);
+				}
+				else
+				{
+					soundEngine.Play2D(sounds[name],
```

Comment:
You can avoid duplication here by pulling out a single `var sound = soundEngine.Play2D(...)` before doing the `currentSounds` manipulation.